modules/bastion/main.tf

```diff
@@ -1,11 +1,84 @@
-resource "azurerm_subnet" "graphdb-bastion-subnet" {
+resource "azurerm_subnet" "graphdb-bastion" {
   name                 = "AzureBastionSubnet"
   resource_group_name  = var.resource_group_name
   virtual_network_name = var.virtual_network_name
   address_prefixes     = var.bastion_subnet_address_prefix
 }
 
-resource "azurerm_public_ip" "graphdb-bastion-public-ip" {
+resource "azurerm_network_security_group" "graphdb-bastion" {
+  name                = "${var.resource_name_prefix}-bastion"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+
+  # The following rules are required by Azure Bastion in order to function properly
+  # See https://learn.microsoft.com/en-us/azure/bastion/bastion-nsg
+  # See https://learn.microsoft.com/en-gb/azure/bastion/native-client#secure
+
+  # INBOUND
+
+  security_rule {
+    name                       = "AllowHTTPSInternetInbound"
+    description                = "Allows specified CIDRs to access the Bastion subnet"
+    priority                   = 100
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_address_prefixes    = var.bastion_allowed_cidr_blocks
+    source_port_range          = "*"
+    destination_address_prefix = "*"
+    destination_port_range     = 443
+  }
+
+  security_rule {
+    name                       = "AllowGatewayManagerInbound"
+    description                = "Allows Gateway Manager to perform system operations"
+    priority                   = 110
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_address_prefix      = "GatewayManager"
+    source_port_range          = "*"
+    destination_address_prefix = "*"
+    destination_port_range     = 443
+  }
+
+  # OUTBOUND
+
+  security_rule {
+    name                       = "AllowSSHOutbound"
+    description                = "Allows outbound SSH/RDP connections"
+    priority                   = 100
+    direction                  = "Outbound"
+    access                     = "Allow"
+    protocol                   = "*"
+    source_address_prefix      = "*"
+    source_port_range          = "*"
+    destination_address_prefix = "VirtualNetwork"
+    destination_port_ranges    = [22, 3389]
```

Comment:
do really need RDP?

Reply:
You cannot associate the subnet otherwise, Azure has some validation under da hood..